File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int N = t.length();
        int M = p.length();
        long num = Long.valueOf(p);
        
        for (int i = 0; i < N-M+1; i++) {
            String tmp = t.substring(i, i+M);
            long tmpNum = Long.valueOf(tmp);
            if (tmpNum <= num) {
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    N = len(t)
    M = len(p)
    for i in range(N-M+1):
        tmp = int(t[i:i+M])
        # print(tmp)
        if int(tmp) <= int(p):
            answer += 1
    return answer